modules/ITK/cli/statismo-build-models-utils.h

```diff
@@ -0,0 +1,175 @@
+        landmarkBasedTransformInitializer->SetFixedLandmarks(fixedLandmarks);
+        landmarkBasedTransformInitializer->SetMovingLandmarks(movingLandmarks);
+        typename TransformType::Pointer transform = TransformType::New();
+        transform->SetIdentity();
+        landmarkBasedTransformInitializer->SetTransform(transform);
+        landmarkBasedTransformInitializer->InitializeTransform();
+
+        typename FilterType::Pointer filter = FilterType::New();
+        filter->SetInput(movingMesh);
+        filter->SetTransform(transform);
+        filter->Update();
+
+        *it = filter->GetOutput();
+    }
+    return translatedMeshes;
+}
+
+
+template<class MeshType>
+float calculateMeshDistance(typename MeshType::Pointer mesh1, typename MeshType::Pointer mesh2) {
+    if (mesh1->GetNumberOfPoints() != mesh2->GetNumberOfPoints() || mesh1->GetNumberOfCells() != mesh2->GetNumberOfCells()) {
+        itk::ExceptionObject e(__FILE__, __LINE__, "Both meshes must have the same number of Edges & Vertices", ITK_LOCATION);
+        throw e;
+    }
+
+    float fDifference = 0;
+    for (unsigned j = 0; j < mesh1->GetNumberOfPoints(); j++) {
+        typename MeshType::PointType point1 = mesh1->GetPoint(j);
+        typename MeshType::PointType point2 = mesh2->GetPoint(j);
+        for (unsigned k = 0; k < mesh1->GetPoint(j).Size(); k++) {
+            fDifference += (point1[k] - point2[k])*(point1[k] - point2[k]);
+        }
+    }
+    fDifference /= (mesh1->GetNumberOfPoints() + MeshType::PointDimension);
+    return fDifference;
+}
+
+template<class MeshType>
+typename MeshType::Pointer calculateMeanMesh(vector<typename MeshType::Pointer> meshes) {
+    if (meshes.size() == 0) {
+        itk::ExceptionObject e(__FILE__, __LINE__, "Can't calculate the mean since no meshes were provided.", ITK_LOCATION);
+        throw e;
+    }
+
+    typename MeshType::Pointer meanMesh;
+    for (typename vector<typename MeshType::Pointer>::const_iterator i = meshes.begin(); i != meshes.end(); i++) {
+        typename MeshType::Pointer mesh = *i;
+        if (!meanMesh) {
+            meanMesh = mesh;
+        } else {
+            if (meanMesh->GetNumberOfPoints() != mesh->GetNumberOfPoints() || meanMesh->GetNumberOfCells() != mesh->GetNumberOfCells()) {
+                itk::ExceptionObject e(__FILE__, __LINE__, "All meshes must have the same number of Edges & Vertices", ITK_LOCATION);
+                throw e;
+            }
+
+            //sum up all meshes
+            for (unsigned j = 0; j < meanMesh->GetNumberOfPoints(); j++) {
+                typename MeshType::PointType point = meanMesh->GetPoint(j);
+                typename MeshType::PointType pointToAdd = mesh->GetPoint(j);
+                for (unsigned k = 0; k < meanMesh->GetPoint(j).Size(); k++) {
+                    point[k] = point[k] + pointToAdd[k];
```

Comment:
it would make a lot of sense to use compensated summation here (Kahan summation),
http://en.wikipedia.org/wiki/Kahan_summation_algorithm
http://www.itk.org/Doxygen46/html/classitk_1_1CompensatedSummation.html